certbot-apache/tests/configurator_test.py

```diff
@@ -1349,6 +1314,20 @@ def test_deploy_cert_not_parsed_path(self):
                 self.assertTrue(mock_add.called)
         shutil.rmtree(tmp_path)
 
+    def test_deploy_cert_no_mod_ssl(self):
+        # Create
+        ssl_vhost = self.config.make_vhost_ssl(self.vh_truth[0])
+        self.config.parser.modules["socache_shmcb_module"] = None
+
+        self.assertRaises(errors.MisconfigurationError, self.config.deploy_cert,
```

Comment:
The error raised here changes depending on the system I run on it on.

On Ubuntu, it's the `"ssl_module" not in self.parser.modules` case in the line you added. On macOS, it's
```
Error running command ['apache2ctl', '-t', '-D', 'DUMP_MODULES'] for runtime parameters!

certbot.errors.MisconfigurationError: Error accessing loaded Apache parameters: ['apache2ctl', '-t', '-D', 'DUMP_MODULES']
```
with the relevant part of the traceback being:
```
> /my/local/path/certbot/certbot-apache/certbot_apache/_internal/configurator.py(596)_deploy_cert()
    595         # cert_key... can all be parsed appropriately
--> 596         self.prepare_server_https("443")
    597 

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/configurator.py(1185)prepare_server_https()
   1184 
-> 1185         self.prepare_https_modules(temp)
   1186         self.ensure_listen(port, https=True)

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/configurator.py(1330)prepare_https_modules()
   1329                 self.parser.aug.load()
-> 1330                 self.parser.reset_modules() # Reset to load the new ssl_module path
   1331                 # Call again because now we can gate on openssl version

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/parser.py(259)reset_modules()
    258         self.modules = {}
--> 259         self.update_modules()
    260         self.parse_modules()

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/parser.py(319)update_modules()
    318 
--> 319         matches = apache_util.parse_modules(self.configurator.option("ctl"))
    320         for mod in matches:

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/apache_util.py(197)parse_modules()
    196                "DUMP_MODULES"]
--> 197     return parse_from_subprocess(mod_cmd, r"(.*)_module")
    198 

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/apache_util.py(210)parse_from_subprocess()
    209     """ 
--> 210     stdout = _get_runtime_cfg(command)
    211     return re.compile(regexp).findall(stdout)

  /my/local/path/certbot/certbot-apache/certbot_apache/_internal/apache_util.py(237)_get_runtime_cfg()
    236             "Error accessing loaded Apache parameters: {0}".format(
--> 237                 command))
    238     # Small errors that do not impede
```
To fix this, what do you think about mocking out `self.config.prepare_server_https` since we're not trying to test any of its behavior in this test.

Reply:
Looks like we can't quite just do that, since that function does other things that the brittle tests rely on (I'm getting a list index out of range error). I'll mock something else.

Reply:
figured it out. please ignore this entire response, I forgot to remove a test.